packages/core/lib/segments/attributes/remote_request_data.js

```diff
@@ -13,8 +13,11 @@ function RemoteRequestData(req, res, downstreamXRayEnabled) {
 }
 
 RemoteRequestData.prototype.init = function init(req, res, downstreamXRayEnabled) {
+  // Not all requests have "agent"
+  const protocol = req.agent ? req.agent.protocol: req.protocol + ':'
```

Comment:
Is it guaranteed that if `req.agent` is absent `req.protocol` will be defined, or should we add an additional check?

Also, is it documented anywhere that `req.protocol` doesn't have a trailing colon but `req.agent.protocol` does? Seems strange that they're different in that way.

Reply:
Good point, `req.protocol` was added in `14.5.0`, so we'll need additional checks.
I haven't seen any documentation, but the source code has the agent protocol defaulting to `https:` https://github.com/nodejs/node/blob/8e173836538efb95f5ac5e898f177efff681d2aa/lib/https.js#L154